vulkaninfo/CMakeLists.txt

```diff
@@ -11,7 +11,35 @@ else()
     add_executable(vulkaninfo vulkaninfo.c)
 endif()
 
-if(APPLE)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    include(FindPkgConfig)
+    option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
+    option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
+    set(VULKANINFO_WSI_SELECTION "XCB" CACHE STRING "Select WSI target for vulkaninfo (XCB, XLIB)")
+
+    if (BUILD_WSI_XCB_SUPPORT)
+        find_package(XCB REQUIRED)
+    endif()
+
+    if (BUILD_WSI_XLIB_SUPPORT)
+        find_package(X11 REQUIRED)
+    endif()
+    
+    if (VULKANINFO_WSI_SELECTION STREQUAL "XCB")
+        if (NOT BUILD_WSI_XCB_SUPPORT)
+            message( FATAL_ERROR "Selected XCB for vulkaninfo build but not building Xcb support" )
+        endif()
+        target_include_directories(vulkaninfo PRIVATE ${XCB_INCLUDE_DIRS})
+        target_link_libraries(vulkaninfo ${XCB_LIBRARIES})
+        target_compile_definitions(vulkaninfo PRIVATE -DVK_USE_PLATFORM_XCB_KHR)
+    elseif (VULKANINFO_WSI_SELECTION STREQUAL "XLIB")
+        if (NOT BUILD_WSI_XLIB_SUPPORT)
+            message( FATAL_ERROR "Selected XLIB for vulkaninfo build but not building Xlib support" )
+        endif()
+        target_include_directories(vulkaninfo PRIVATE ${X11_XCB_INCLUDE_DIRS})
```

Comment:
We want `X11_INCLUDE_DIR` here.  No 'S'.  Please see https://cmake.org/cmake/help/v3.0/module/FindX11.html

> X11_FOUND        - True if X11 is available
> X11_INCLUDE_DIR  - include directories to use X11
> X11_LIBRARIES    - link against these to use X11
>